streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamFlatMapTest.java

```diff
@@ -86,4 +88,10 @@ public void testFlatMap() {
             assertEquals(expected[i], supplier.theCapturedProcessor().processed().get(i));
         }
     }
+
+    @Test
+    public void testKeyValueMapperResultNotNull() {
+        final KStreamFlatMap<String, Integer, String, Integer> supplier = new KStreamFlatMap<>((key, value) -> null);
+        assertThrows(NullPointerException.class, () -> supplier.get().process(new Record<>("K", 0, 0L)));
```

Comment:
As the whole point of this PR is to provide better messages, I would also check in the test that the exception has the new enhanced message. Something like 
```suggestion
        final Record<String, Integer> record = new Record<>("K", 0, 0L);
        assertThrows(NullPointerException.class, () -> supplier.get().process(record), String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```

Reply:
Thanks for your review! I found we mixed the using of two packages: `org.junit.Assert` and `org.junit.jupiter.api.Assertions`.  The method `asssertThrows` in those two packages have different api:

For `org.junit.Assert.assertThrows`, the [api](https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html#assertThrows(java.lang.String,%20java.lang.Class,%20org.junit.function.ThrowingRunnable)) is:

```java
public static <T extends Throwable> T assertThrows(String message,
                                                   Class<T> expectedThrowable,
                                                   ThrowingRunnable runnable)
```

For `org.junit.jupiter.api.Assertions.assertThrows`, the [api](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertThrows-java.lang.Class-org.junit.jupiter.api.function.Executable-java.lang.String-) is:
```java
public static <T extends Throwable> T assertThrows(Class<T> expectedType,
                                                   Executable executable,
                                                   String message)
```

I searched our Kafka code base and found there is [another place](https://github.com/apache/kafka/blob/9565a529e08d7aa36beac02c8e6115bcc87d2dc7/streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java#L521) using the package `org.junit.Assert.assertThrows` to assert exception message so I follow that to use the same apii as well here.

Please let me know if this is not proper.

Reply:
Actually I was wrong with my code suggestion, the overloaded method doesn't do what I expected it to do.
The message it takes is only to print if the assertion fails.

That should do the trick!
```
final Throwable throwable = assertThrows(String.format("KeyValueMapper can't return null from mapping the record: %s", record), NullPointerException.class, () -> supplier.get().process(record));
        assertEquals(throwable.getMessage(), String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```
Sorry about the confusion I brought with the first comment

Reply:
@jlprat  I see. I didn't notice this, either LOL. Fixed!